### PR TITLE
1.0 Beta Release

### PR DESCRIPTION
**Pull request by knicklabs**
This PR prepares everything for a beta release of Brigade 1.0.

Notable changes:

- Remove `default export` for Brigade
- Export `ReactMounter` (formally `default export`) and `ReactView`
- Rename `mixins` to `hof` to signal that mixins were replaced with higher order functions
- Reorganize stories into `ReactMounter` and `ReactView` directories
- Rename `Example` stories to `ReactMounter` stories

In summary, Brigade no longer has default usage. Users need to pick a flavour (`ReactMounter` or `ReactView`) based on their use case. 

`import brigade from '@helpscout/brigade'` becomes `import {ReactMounter} from '@helpscout/brigade'` or `import {ReactView} from '@helpscout/brigade'`.

### CODE REVIEW

**Comment by ItsJonQ** [comment]
> In summary, Brigade no longer has default usage. Users need to pick a flavour (ReactMounter or Reactview) based on their use case.

👍 A good pattern to indicate/encourage a usage path for the user

**Comment by knicklabs** [comment]
@ItsJonQ and @brettjonesdev, 

I've reorganized the project so that ReactMounter and ReactView live side-by-side. I've also made a few adjustments to the storybook and completely overhauled the docs. 

The documentation and storybook websites are currently tracking this branch:

https://brigade.netlify.com/ - 1.0.0 documentation
https://brigade-storybook.netlify.com/ - 1.0.0 storybook

@brettjonesdev I took a stab at the ReactView docs and positioning ReactView use case. Would appreciate your feedback or also feel free to just update the docs if its easier.

Once this is approved, we can release 1.0.0

**Comment by brettjonesdev** [comment]
Looking good! We might note somewhere that Marionette 1.8.8 is a requirement for `ReactView`. We can iterate on these docs though. Nice work!

**Comment by knicklabs** [comment]
@brettjonesdev Good call. I had it in the dependencies section but that's easy to overlook. I added a notice to the `ReactView` page too. 

Screenshot
